### PR TITLE
MGMT-4114 Refactor KubeAPI subsystem and use clusterDeployment

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

File: subsystem/kubeapi_test.go [hunk 1]
```diff
@@ -120,35 +128,55 @@ func waitForClusterCRDState(
...
+					},
+					AgentSelector: metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}},
+				},
+			},
 		},
 	}
 }
 
 func cleanUP(ctx context.Context, client k8sclient.Client) {
 	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:177]: Still needed to delete `v1alpha1.Cluster{}`?

Comment (nmagnezi) [reply subsystem/kubeapi_test.go:177]: Nop

File: subsystem/kubeapi_test.go [hunk 2]
```diff
@@ -102,15 +103,22 @@ func getClusterCRD(ctx context.Context, client k8sclient.Client, key types.Names
 	return cluster
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:103]: Need to update getClusterCRD

File: subsystem/kubeapi_test.go [hunk 3]
```diff
@@ -87,30 +88,37 @@ func getClusterFromDB(
...
+		//Expect(clusterCRD.Status.Error).Should(Equal(""))
 		time.Sleep(time.Second)
 	}
 	Expect(err).To(BeNil())
 	return cluster
 }
 
-func getClusterCRD(ctx context.Context, client k8sclient.Client, key types.NamespacedName) *v1alpha1.Cluster {
-	cluster := &v1alpha1.Cluster{}
+//func getClusterCRD(ctx context.Context, client k8sclient.Client, key types.NamespacedName) *v1alpha1.Cluster {
```
Comment (nmagnezi) [inline subsystem/kubeapi_test.go:99]: to be removed

File: subsystem/kubeapi_test.go [hunk 4]
```diff
@@ -204,6 +221,6 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}
 		generateFullMeshConnectivity(ctx, "1.2.3.10", hosts...)
 
-		waitForClusterCRDState(ctx, kubeClient, key, models.ClusterStatusPreparingForInstallation, waitForClusterReconcileTimeout)
+		waitForClusterDeploymentCRDState(ctx, kubeClient, key, models.ClusterStatusReady, waitForClusterReconcileTimeout)
```
Comment (nmagnezi) [inline subsystem/kubeapi_test.go:224]: @RazRegev let me know if changing to `ready` makes sense

File: subsystem/kubeapi_test.go [hunk 5]
```diff
@@ -120,34 +121,50 @@ func waitForClusterCRDState(
...
+						ControlPlaneAgents: 3,
+						WorkerAgents:       0,
+					},
+				},
+			},
 		},
+		Platform: hivev1.Platform{
+			AgentBareMetal: &agentv1.BareMetalPlatform{
+				APIVIP:            "1.2.3.8",
+				APIVIPDNSName:     "test.domain",
```
Comment (filanov) [inline subsystem/kubeapi_test.go:157]: Need to remove this, it's relevant only for day2 clusters. 
This is what makes your test fail.Update cluster ignore it so `updateIfNeeded` always see a difference and try to update it again, there is no error in the logs for it.

**Review by filanov — CHANGES_REQUESTED** [review]

File: subsystem/kubeapi_test.go (hunk 4 above)
Comment (filanov) [reply subsystem/kubeapi_test.go:224]: It's not. need to be `ClusterStatusPreparingForInstallation`

**Comment by nmagnezi** [comment]
Test service log [here](https://gist.github.com/nmagnezi/2a08df78c77ffcffe6b2edf6b085dd79)

**Review by filanov — APPROVED** [review]

**Pull request merged**